service/src/test/java/com/myhome/services/unit/MailSDJpaServiceTest.java

```diff
@@ -1,39 +1,64 @@
 package com.myhome.services.unit;
 
+import com.myhome.domain.SecurityToken;
 import com.myhome.domain.User;
 import com.myhome.services.springdatajpa.MailSDJpaService;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.mockito.InjectMocks;
 import org.mockito.Mock;
 import org.mockito.MockitoAnnotations;
-import org.springframework.mail.MailException;
-import org.springframework.mail.SimpleMailMessage;
+import org.springframework.context.support.ResourceBundleMessageSource;
+import org.springframework.mail.MailSendException;
 import org.springframework.mail.javamail.JavaMailSender;
+import org.springframework.test.util.ReflectionTestUtils;
+import org.thymeleaf.ITemplateEngine;
+import org.thymeleaf.context.Context;
+
+import javax.mail.Session;
+import javax.mail.internet.MimeMessage;
 
 import static org.junit.jupiter.api.Assertions.assertFalse;
 import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.BDDMockito.given;
 import static org.mockito.Mockito.doThrow;
-import static org.mockito.Mockito.mock;
 
 class MailSDJpaServiceTest {
 
   @Mock
   private JavaMailSender mailSender;
+  @Mock
+  private ITemplateEngine emailTemplateEngine;
+  @Mock
+  private ResourceBundleMessageSource messageSource;
 
   @InjectMocks
   private MailSDJpaService mailSDJpaService;
 
   @BeforeEach
   private void init() {
     MockitoAnnotations.initMocks(this);
+    ReflectionTestUtils.setField(mailSDJpaService, "username", "username");
```

Comment:
W definitely don't want to use reflection here :)

Reply:
I cant find any other way to do this without reflection. `MailSDJpaService` is bean and Spring must set up it's fields but unit tests runs without `SpringContext` so how can I set fields without constructing instance `MailSDJpaService` manualy?

Reply:
And I dont think thats default values for all fields is good idea

Reply:
You created the class `MailProperties`, now you can create the `mailSDJpaService` via constructor in tests (omitting the @InjectMocks annotation)

Reply:
`MailSDJpaService mailService = new MailSDJpaService(emailTemplateEngine, mailSender, messageSource, mailProperties)`

In that case all of `emailTemplateEngine`, `mailSender`, `messageSource` can be used as mocked and `mailProperties` can be created via new in tests with appropriate values.